index.rst

```diff
@@ -1,60 +1,850 @@
+So I'll just leave most of the content as-is, and list here the big ideas I'd want to explore further:
+
+- Creating a SQLite staging registry (even for local execution), by transferring input datasets needed from an "upstream" shared registry after `~lsst.pipe.base.QuantumGraph` generation.
+  Datasets would only be pushed back to the upstream repo if explicitly requested.
+  This could be a one-step way to provide a lot more insulation between users, and hence make the shared repo access model easier to implement (maybe the only non-admin write operations we need to support are ingest and import?) while also giving users more power to play in their own sandboxes.
+  It also opens up a lot of intriguing possibilities for campaign features (more on that in later bullets).
+  I think the big complications this raises are about how to handle multiple transfers from upstream (if needed), when the staging repo doesn't start out empty and there's potential for conflicts, especially when `~lsst.pipe.base.QuantumGraph` generation is run multiple times over the life of the campaign.
+
+- I don't think I've tried quite hard enough to remove functionality here that isn't obviously needed in the name of simplification.
+  One concrete example is that I don't think anyone really needs to be able to change the input collections after a campaign is created, though it's also impossible to rigorously guard against the possibility that input collection *content* may have changed (unless we only care about changes to a local staging repo!), so that may not be as much of a simplification as I hope.
+
+- We could at least sometimes initialize a git repository in the campaign directory and commit every time we run ``pipetask``.
+  That would provide a really nice history-tracking and rollback feature almost for free, though it's not quite as useful if we don't also go in the direction of having a local SQLite registry that we could also git-control, and it doesn't seem like something that's viable for large campaigns (unless we can define a sound subset of the campaign content that's still worth putting in git).
+
+- Michelle Gower brought up the possibility of putting the campaign metadata into the registry database, as that's desirable for long-term control and management of that at least in the BPS context.
+  I'm a bit wary of expanding ``butler``/``pipetask`` to centrally control even more things (developer campaigns may be quite different from production campaigns in that respect - sandboxes vs. data releases).
+  But combined with a local SQLite repo, so the metadata for sandbox campaigns doesn't get pushed upstrea until/unless a run is pushed back, it's very intriguing.
+
+- The possibility of putting log files in the Registry as well was also raised by Michelle.
```

Comment:
I'm a bit worried about this.  If there's anything in the logs that is required to be preserved as part of the permanent survey record, that suggests to me that provenance or metric persistence is inadequate.  On the other hand, they are relatively cheap insurance against such inadequacy.  (But note daunting Qserv experience.)

Reply:
I think the sentiment was more about "we shouldn't invent _another_ system for relating datasets to runs", rather than "required to be preserved", but I should let @MichelleGower speak for herself on that.